engine/src/test/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutputTest.java

```diff
@@ -417,6 +417,7 @@ private List<Throwable> doOutput( TextFileField[] textFileField, List<Object[]>
       .calcMetaWithFieldOptions( any( TextFileOutputData.class ) );
     when( stepMockHelper.processRowsStepMetaInterface.getMetaWithFieldOptions() ).thenCallRealMethod();
 
+    stepMockHelper.processRowsStepMetaInterface.metaWithFieldOptions = null;
```

Comment:
Add a comment in here regarding why this is necessary; the step meta code is being executed multiple times without being fully re-initialized, and the step code expects this field to be null when processing the first row during execution.